### PR TITLE
add --fingerprint option to chia keys show

### PR DESCRIPTION
**Pull request by jack60612**
this pr adds a --fingerprint option to chia keys show

```
(venv) PS C:\Users\jackm\GitHub\chia-blockchain> chia keys show --fingerprint 1116336227
Showing all public keys derived from your master seed and private key:

Label: test
Fingerprint: 1116336227
Master public key (m): a09cef259f31c5daaf3a0d237033721fb746075c6d809245103affb02a8bfaed3e2930d86e9e66a2e7a2fcb4b67a3f5c
Farmer public key (m/12381/8444/0/0): b094f390a391d10eda509f3f8fade8e3e432a6f0d1ad078108814fdfc43f81f6efd714692450c4f75228b9746eef4095
Pool public key (m/12381/8444/1/0): 81fcb6d215887f694516d3d478253f61e017d20b06b4a7d761c5e63775ce25ec56ca6f9f1adc737a75f38eefd2305fbc
First wallet address: txch17papy3kfhrx7cl6xupnhgdwkf992hqlunpv4sggjmfcykel4nx5snwkt4t

```

### CODE REVIEW

File: chia/cmds/keys_funcs.py [hunk 1]
```diff
@@ -126,13 +126,18 @@ def delete_key_label(fingerprint: int) -> None:
         sys.exit(f"Error: {e}")
 
 
-def show_all_keys(root_path: Path, show_mnemonic: bool, non_observer_derivation: bool, json_output: bool):
+def show_all_keys(
```
Comment (aqk) [inline chia/cmds/keys_funcs.py:129]:
```suggestion
def show_keys(
```
No need to change this - just wanted to point it out. Feel free to ignore this suggestion.

Comment (jack60612) [reply chia/cmds/keys_funcs.py:129]: > No need to change this - just wanted to point it out. Feel free to ignore this suggestion.

makes more sense

**Review by aqk — DISMISSED** [review]

**Review by jack60612 — COMMENTED** [review]
I need to add cli tests

File: tests/core/cmds/test_keys.py [hunk 2]
```diff
@@ -378,6 +378,35 @@ def test_show(self, keyring_with_one_key, tmp_path):
...
+        ]
+        runner = CliRunner()
+        cmd_params = ["keys", "show", "--fingerprint", TEST_FINGERPRINT]
+        # Generate a new config
+        assert runner.invoke(cli, [*base_params, "init"]).exit_code == 0
+        # Run the command
+        result: Result = runner.invoke(cli, [*base_params, *cmd_params])
+
+        # assert result.exit_code == 0
+        assert len([word for word in result.output.split() if word == "Fingerprint:"]) == 1
```
Comment (xdustinface) [inline tests/core/cmds/test_keys.py:408]:
```suggestion
        assert result.exit_code == 0
        fingerprints = [line for line in result.output.splitlines() if "Fingerprint:" in line]
        assert len(fingerprints) == 1
        assert str(TEST_FINGERPRINT) in fingerprints[0]
```

To uncomment the `assert result.exit_code == 0` and also test if the expected fingerprint was returned.

**Review by xdustinface — CHANGES_REQUESTED** [review]

**Review by xdustinface — APPROVED** [review]

**Review by emlowe — APPROVED** [review]

**Pull request merged**